packages/mattermost-redux/src/selectors/entities/channels.ts

```diff
@@ -879,6 +922,109 @@ export const getUnreadChannels: (state: GlobalState, lastUnreadChannel?: Channel
+            const c = channels[id];
+
+            if (c.type === General.DM_CHANNEL || c.type === General.GM_CHANNEL) {
+                return completeDirectChannelDisplayName(currentUser.id, profiles, userIdsInChannels[id], settings!, c);
+            }
+
+            return c;
+        });
+    },
+);
+
+export const sortUnreadChannels = (
+    channels: Channel[],
+    myMembers: RelationOneToOne<Channel, ChannelMembership>,
+    lastPosts: RelationOneToOne<Channel, Post>,
+    crtEnabled: boolean,
+    lastUnreadChannel?: LastUnreadChannel | null,
+) => {
+    function isMuted(channel: Channel) {
+        return isChannelMuted(myMembers[channel.id]);
+    }
+
+    function hasMentions(channel: Channel) {
+        if (lastUnreadChannel && channel.id === lastUnreadChannel.id && lastUnreadChannel.hadMentions) {
+            return true;
+        }
+
+        const member = myMembers[channel.id];
+        return member?.mention_count !== 0;
+    }
+
+    // Sort channels with mentions first and then sort by recency
+    return [...channels].sort((a, b) => {
+        // Sort muted channels last
+        if (isMuted(a) && !isMuted(b)) {
+            return 1;
+        } else if (!isMuted(a) && isMuted(b)) {
+            return -1;
+        }
+
+        // Sort non-muted mentions first
+        if (hasMentions(a) && !hasMentions(b)) {
+            return -1;
+        } else if (!hasMentions(a) && hasMentions(b)) {
+            return 1;
+        }
+
+        let lastPostAt = {
+            a: a.last_post_at,
+            b: b.last_post_at,
+        };
+
+        if (crtEnabled) {
+            lastPostAt = {
+                a: a.last_root_post_at,
+                b: b.last_root_post_at,
+            };
+        }
+
+        // If available, get the last post time from the loaded posts for the channel, but fall back to the
+        // channel's last_post_at if that's not available. The last post time from the loaded posts is more
+        // accurate because channel.last_post_at is not updated on the client as new messages come in.
+        const aLastPostAt = maxDefined(lastPostAt.a, lastPosts[a.id]?.create_at);
+        const bLastPostAt = maxDefined(lastPostAt.b, lastPosts[b.id]?.create_at);
+
+        return bLastPostAt - aLastPostAt;
+    });
+};
+
+export const getSortedAllTeamsUnreadChannels: (state: GlobalState, lastUnreadChannel?: LastUnreadChannel | null) => Channel[] = createSelector(
+    'getSortedAllTeamsUnreadChannels',
+    getUnsortedAllTeamsUnreadChannels,
+    getMyChannelMemberships,
+    getLastPostPerChannel,
+    (state: GlobalState, lastUnreadChannel: LastUnreadChannel | undefined | null = null): LastUnreadChannel | undefined | null => lastUnreadChannel,
+    isCollapsedThreadsEnabled,
+    (channels, myMembers, lastPosts, lastUnreadChannel, crtEnabled) => {
+        return sortUnreadChannels(channels, myMembers, lastPosts, crtEnabled, lastUnreadChannel);
```

Comment:
Is there a reason that you swapped the order of those last two arguments? I think it would be safer if we kept those consistent since people might mix those up down the line

Reply:
Yeah, I made ``lastUnreadChannel`` optional, that’s why but no issue will keep the order same and pass ``null`` as you suggested earlier.